_posts/2019/2019-09-22-a-rails-active-model-validator-for-stripe-ids.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "A Active Model validator for Stripe Ids"
+description: "A decent example"
+layout: article
+category: ruby
+image:
+  base: '2019/a-rails-active-model-validator-for-stripe-ids'
+  alt: 'Zebra stripes'
+  credit: David Clarke
+  source: "https://unsplash.com/photos/sN6d60TySV0"
+
+---
+
+In [my article on using validators](/ruby/use-a-custom-validator), the example was pretty simple.
+
+This is an example of real production use of a more complex validator extracted from the billing code in [AnswerThePublic](https://answerthepublic.com). We use [Stripe](https://stripe.com) exclusively for the billing in our applications.
+
+
+## Instead of...
+
+...using a regex in `validates` when setting a format for the `stripe_id`:
+
+```ruby
+class Subscription < ActiveRecord
+  validates :stripe_id, format: { with: /\Asub\_[A-Za-z0-9]{8,}\z/ }
+end
+```
+
+
+## Use...
+
+...a validator class to ensure the format of a Stripe `id`...
```

Comment:
This doesn't read so well. And it's not clear what a `Stripe id` is at this point, to someone who hasn't used Stripe. I haven't used it in a while and it took me a second to work out what you were talking about.